### ISSUE TITLE
Calendar events wrong time (timezone issues?)

### ISSUE THREAD

**Issue by pauby** [muness/obsidian-ics#37]
When importing the calendar some events are appearing at the wrong times. These appear to be meetings created in a different timezone to my own.

Is there handling for this? And is there anything I can do troubleshoot and provide more info?

**Comment by efbenson** [comment]
It looks like its an issue with reoccuring events created in another timezone I am seeing it as well.  The logic for parsing the time for reoccuring is not taking account of the tz

https://github.com/muness/obsidian-ics/blob/f3d796df3ee828517da33d9ef762880398e0ae98/src/icalUtils.ts#L76

from https://www.npmjs.com/package/node-ical:

``` Javascript
       if (event.rrule.origOptions.tzid) {
                // tzid present (calculate offset from recurrence start)
                const dateTimezone = moment.tz.zone('UTC')
                const localTimezone = moment.tz.guess()
                const tz = event.rrule.origOptions.tzid === localTimezone ? event.rrule.origOptions.tzid : localTimezone
                const timezone = moment.tz.zone(tz)
                const offset = timezone.utcOffset(date) - dateTimezone.utcOffset(date)
                newDate = moment(date).add(offset, 'minutes').toDate()
            } else {
                // tzid not present (calculate offset from original start)
                newDate = new Date(date.setHours(date.getHours() - ((event.start.getTimezoneOffset() - date.getTimezoneOffset()) / 60)))
            }
``` 

You want to apply that logic to the reoccuring event.

**Comment by muness** [comment]
I tried that logic, and it looks like it's buggy based on walking through it. I see the logseq plugin also has the logic commented out: https://github.com/sawhney17/logseq-calendars-plugin/blob/5524a523a861280d04695f1aafc95c4275a5bbbd/index.ts#L210

As it stands this is applying the offset from UTC rather than the local timezone

**Comment by muness** [comment]
I think this is now fixed with #47 . Please reopen if you're still seeing the issue.
